### PR TITLE
UI/contactnsubs

### PR DESCRIPTION
**Pull request by hassanabbas16**
**Assigned Tasks:**

**Booking Form Responsiveness:**
General:
1. Step Bar:
- Made the step bar item container be transparent as per the themes
- Made the checkmark icon to a tick icon when on small screens
- Made the step bar items smaller and scale when the user is on the current step.
- Made the step bar width responsive

2. Price and Buttons:
- Made the price buttons widths and fonts responsive as per to screen width on phones
- Made the Price tag and line be responsive to screen width

3. Toggle Switch:
- Made the toggle switch fit height and width as per specifiec in the figma
- Made its fonts responsive to screen width and theme mode

Step 4:
- Made the packages containers fit the figma specifications
- Implemented a dynamic swipper to allow user to see different packages options
- Allowed pagination and basic highlighting for the cards

Step 5:
- Implemented the swipper functionality same as above
- Made the height of the swipper dynamic to fit all content for all 3 types of packages.
- Made the cards as per the figma specifications and design

Step 6:
- Made the widths of the options responsive to screen width
- Made the heights and headings and the options to represent the package selected with their specific colors
- Made the whole step responsive to dark mode

Step 7:
- Made the widths of the options responsive to screen width
- Made the heights and headings and the options to represent the package selected with their specifc colors
- Made the whole step responsive to dark mode

Step 8:
- Implemented the date calender to incorporate figma design
- Allow user to select dates from a custom modal when a specific date is selected

Step 9:
- Made the summary sections responsive to screen width
- Adjusted the overflowing sections and the fontsizes

Step 10:
- Rendered the form as per figma design for responsiveness
- Made the input boxes less rounder and fixed the font sizes

**Screenshots**
screencapture-localhost-3000-booking-2024-10-08-10_37_49
screencapture-localhost-3000-booking-2024-10-08-10_38_01
screencapture-localhost-3000-booking-2024-10-08-10_31_01
screencapture-localhost-3000-booking-2024-10-08-10_31_31
screencapture-localhost-3000-booking-2024-10-08-10_31_49
screencapture-localhost-3000-booking-2024-10-08-10_35_14
screencapture-localhost-3000-booking-2024-10-08-10_35_54

### CODE REVIEW

File: src/app/subscribe/page.jsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ const PackageCard = ({ pkg, index, highlightColor }) => {
     return (
         <StyledCard>
             <StyledImageContainer highlightColor={highlightColor} theme={theme}>
-                <img src={`/sub${index + 1}.png`} alt={`${pkg.name} image`} style={{ width: '460px', height: '270px', objectFit: 'cover', boxShadow: theme.palette.mode === 'light' ? '0px 4px 30.1px rgba(0, 0, 0, 0.5)' : 'none', }} />
+                <img src={`/bookingFormIcons/sub${index + 1}.png`} alt={`${pkg.name} image`} style={{ width: '460px', height: '270px', objectFit: 'cover', boxShadow: theme.palette.mode === 'light' ? '0px 4px 30.1px rgba(0, 0, 0, 0.5)' : 'none', }} />
```
Comment (anasbinarif) [inline src/app/subscribe/page.jsx:36]: why are you using style here? also, try to move the style to a separate file if its more than 4 5 lines in the whole code. contact Faizan if you need any help

File: src/app/subscribe/page.jsx [hunk 2]
```diff
@@ -54,7 +54,7 @@ const PackageCard = ({ pkg, index, highlightColor }) => {
             <StyledOptionsList theme={theme}>
                 {pkg.packages.map((item) => (
                     <Box key={item}>
-                        <img src="/Checkmark.png" alt="Checkmark" />
+                        <img src="/bookingFormIcons/Checkmark.png" alt="Checkmark" />
```
Comment (anasbinarif) [inline src/app/subscribe/page.jsx:57]: use Image from nextjs

File: src/components/BookingForm/PersonParticularsStep/BookingParticulars.jsx [hunk 3]
```diff
@@ -9,25 +9,6 @@ import React, { useEffect, useState } from "react";
 import { useValidation } from "../../../contexts/ValidationContext";
 import { useSession } from "next-auth/react";
 
-export const FormTwoColumn = styled(Box)(({ theme }) => ({
```
Comment (anasbinarif) [inline src/components/BookingForm/PersonParticularsStep/BookingParticulars.jsx:12]: move this to .style file, if you don't know about this text me on slack we will have aa short call

File: src/components/BookingForm/SubscriptionPackagesStep/SubscriptionPackages.jsx [hunk 4]
```diff
@@ -64,7 +52,31 @@ const SubscriptionPackages = () => {
...
     <SubscriptionPkgsContainer isMobile={isMobile}>
       {isMobile ? (
-        <Swiper spaceBetween={0.5} slidesPerView={2.2} style={{ width: "100%" }}>
+          <Swiper
+              modules={[Pagination]}
+              slidesPerView="1.75"
+              centeredSlides={true}
+              spaceBetween={14}
+              pagination={{ clickable: true }}
+              breakpoints={{
```
Comment (anasbinarif) [inline src/components/BookingForm/SubscriptionPackagesStep/SubscriptionPackages.jsx:61]: why are breakpoints defined here and not in the create theme globally?

File: src/components/BookingForm/SummaryStep/Summary.jsx [hunk 5]
```diff
@@ -3,20 +3,14 @@ import { Box } from "@mui/system";
 import React from "react";
 import useMultiStepForm from "../../../hooks/useMultiStepForm";
 import { useValidation } from "../../../contexts/ValidationContext";
-
-const SummaryHeading = styled(Typography)(({ theme }) => ({
```
Comment (anasbinarif) [inline src/components/BookingForm/SummaryStep/Summary.jsx:7]: move this to separate .style file

File: src/components/Home/ServicesOverview/ServiceOverviewPckgs.jsx [hunk 6]
```diff
@@ -62,3 +62,16 @@ export const ServiceItemDescription = styled(Typography)(({ theme }) => ({
   fontWeight: "300",
   color: theme.palette.mode === "dark" ? "#C2C2C2" : "#535353",
 }));
+
+export const ContactCardContainer = styled(Box)(({theme}) => ({
```
Comment (anasbinarif) [inline src/components/Home/ServicesOverview/ServiceOverviewPckgs.jsx:66]: this too

**Review by anasbinarif — CHANGES_REQUESTED** [review]

**Comment by anasbinarif** [comment]
also kindly resolve deployment errors

**Comment by fizo-neechan** [comment]
@hassanabbas16 For the last time man, make your PR properly with titles and descriptions that reflect what the the task was and how you have achieved it.

**Comment by fizo-neechan** [comment]
And for the love of god, use a formatter like prettier or whatever and use a linter. Why am i getting these kinds of import errors [1db4512](https://github.com/SaadBJavaid/Fast-Clean-Service/pull/15/commits/1db45123cb248f57fb550336e4cf9f35dc204c0f)

**Review by anasbinarif — CHANGES_REQUESTED** [review]
kindly recheck the commits in this pr, it has faizan commits too

**Comment by anasbinarif** [comment]
the deployment got failed

**Pull request merged**